internal/e2e/metric_stability_test.go

```diff
@@ -0,0 +1,366 @@
+		if !found {
+			t.Logf("expected metric %s was not found", metricName)
+			return false
+		}
+	}
+
+	return true
+}
+
+func TestMetricStability(t *testing.T) {
+	tests := []struct {
+		name            string
+		configFile      string
+		expectedMetrics map[string]bool
+		otelPort        string
+		metricsPort     string
+	}{
+		{
+			name:       "No metric readers (default)",
+			configFile: "metric_stability_test_no_readers.yaml",
+			expectedMetrics: map[string]bool{
+				// Process metrics
+				"otelcol_process_uptime":                         false,
+				"otelcol_process_cpu_seconds":                    false,
+				"otelcol_process_memory_rss":                     false,
+				"otelcol_process_runtime_heap_alloc_bytes":       false,
+				"otelcol_process_runtime_total_alloc_bytes":      false,
+				"otelcol_process_runtime_total_sys_memory_bytes": false,
+
+				// Batch processor metrics
+				"otelcol_processor_batch_batch_send_size":       false,
+				"otelcol_processor_batch_batch_send_size_bytes": false,
+				"otelcol_processor_batch_metadata_cardinality":  false,
+				"otelcol_processor_batch_timeout_trigger_send":  false,
+
+				// HTTP server metrics
+				"http_server_request_body_size":  false,
+				"http_server_request_duration":   false,
+				"http_server_response_body_size": false,
+
+				// Exporter metrics
+				"otelcol_exporter_sent_metric_points":        false,
+				"otelcol_exporter_send_failed_metric_points": false,
+				"otelcol_exporter_sent_spans":                false,
+				"otelcol_exporter_send_failed_spans":         false,
+				"otelcol_exporter_sent_log_records":          false,
+				"otelcol_exporter_send_failed_log_records":   false,
+
+				// Receiver metrics
+				"otelcol_receiver_accepted_metric_points": false,
+				"otelcol_receiver_refused_metric_points":  false,
+				"otelcol_receiver_accepted_spans":         false,
+				"otelcol_receiver_refused_spans":          false,
+				"otelcol_receiver_accepted_log_records":   false,
+				"otelcol_receiver_refused_log_records":    false,
+
+				// Other metrics
+				"promhttp_metric_handler_errors_total": false,
+				"target_info":                          false,
+			},
+			otelPort:    getFreePort(t),
+			metricsPort: "8888", // default metrics port
+		},
+		{
+			name:       "Metric readers",
+			configFile: "metric_stability_test_readers.yaml",
+			expectedMetrics: map[string]bool{
+				// Process metrics
+				"otelcol_process_uptime_seconds_total":            false,
+				"otelcol_process_cpu_seconds_total":               false,
+				"otelcol_process_memory_rss_bytes":                false,
+				"otelcol_process_runtime_heap_alloc_bytes":        false,
+				"otelcol_process_runtime_total_alloc_bytes_total": false,
+				"otelcol_process_runtime_total_sys_memory_bytes":  false,
+
+				// Batch processor metrics
+				"otelcol_processor_batch_batch_send_size":            false,
+				"otelcol_processor_batch_batch_send_size_bytes":      false,
```

Comment:
Alright, that looked weird, but looking at the [metadata.yaml file](https://github.com/open-telemetry/opentelemetry-collector/blob/8b73477b981084947edbb878957e1976707484b2/processor/batchprocessor/metadata.yaml#L29-L42), there really are two metrics with "same" name, just changing that the unit was added as suffix.

From my understanding, [this goes against OTel's spec for metric names](https://opentelemetry.io/docs/specs/semconv/general/metrics/#units:~:text=Conventional%20metrics%20or%20metrics%20that%20have%20their%20units%20included%20in%20OpenTelemetry%20metadata%20(e.g.%20metric.WithUnit%20in%20Go)%20SHOULD%20NOT%20include%20the%20units%20in%20the%20metric%20name.). Let's worry about that in the right time :P